src/Behat/Behat/Context/Cli/InteractiveContextIdentifier.php

```diff
@@ -57,6 +57,10 @@ public function __construct(TranslatorInterface $translator, InputInterface $inp
      */
     public function guessTargetContextClass(ContextEnvironment $environment)
     {
+        if ($this->input->hasParameterOption('--no-interaction')) {
```

Comment:
you should use ``$input->isInteractive()`` instead. The input can be non-interactive without having this option, thanks to autodetection of whether the command is run in a TTY or no.